packages/rum/src/domain/record/observers.ts

```diff
@@ -371,9 +371,41 @@ function initStyleSheetObserver(cb: StyleSheetRuleCallback): ListenerHandler {
     },
   })
 
+  const { stop: restoreInsertNestedRule } = instrumentMethodAndCallOriginal(CSSGroupingRule.prototype, 'insertRule', {
+    before(rule, index) {
+      if (hasSerializedNode(this.parentStyleSheet!.ownerNode!)) {
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏don't assume that a CSSGroupingRule has a stylesheet, because it might not be inserted

![Screenshot 2022-08-19 at 16 26 01](https://user-images.githubusercontent.com/61787/185640210-1275eba2-4906-489c-bc94-5a4a8579c3c5.png)
.